### DISCUSSION TITLE
FR - Add Breadcrumbs and/or Parent Column to Structure Search/Filter

### DISCUSSION THREAD

**Discussion by jacobgraf** [craftcms/cms#12666] [Ideas] [answered]
I am having a UI/UX issue with Structure sections. This mainly has to do with searching and filtering. My issue is that I have multiple entries that have the same name, but that fall under different parents in the Structure section. (e.g. https://cln.sh/GfPkVMGVrbCSjcq1ZCdq) When I search, or filter entries (within an entries field condition) I have no idea what entry is what, because I can't see the breadcrumb. In my case, I really need to know what parent each entry falls under. (e.g. https://cln.sh/rkp2lsWTC8wwVJQZgJXt)

I know there isn't native functionality to fix this (unless you have a clever work-around?), but I believe this could be fixed by adding a "Parent" and/or "Breadcrumb" option to the search/filter view settings so I could easily see which entry is which. (e.g. https://cln.sh/M7JpMX8jW4DdfqF9D4H6)

Any help you can give me would be appreciated. As-is, I need to randomly pick one, and then open it within the entry to see which one it is and it's a huge PITA.

Thanks!

**Comment by brandonkelly** [comment, marked as answer]
Just added a new “Parent” column option to Entry and Category indexes for Craft 4.4, which can be added via the “View” menu whenever needed. (c09456679b0964b2aa41c9cee631b5f8fd5887b5)

**Update:** Also added a new “Ancestors” column option as well. (7fafcf9743b77469a14299b18866771a0de610e2)

<img width="454" alt="Category search results, with an “Ancestors” column" src="https://user-images.githubusercontent.com/47792/218607373-c95940e6-20e5-429b-a2f3-2899bccf7f6a.png">

**Reply by jacobgraf** [reply]
> Just added a new “Parent” column option to Entry and Category indexes for Craft 4.4, which can be added via the “View” menu whenever needed. (c09456679b0964b2aa41c9cee631b5f8fd5887b5)

Awesome! Thanks! This is a great start but it probably won't solve the case where my 2nd level Structure entries are the same and I need to ultimately know what the first level is.

This could be solved 2 ways. A breadcrumb view field or a Top Level Parent view option.

In my opinion, adding a Breadcrumb view option that would be best.

Another elegant solution would simply be to add a full structure breadcrumb hover text over the Entry link.

Hope the feedback helps!

**Reply by brandonkelly** [reply]
Just added a new “Ancestors” column as well, which should fully resolve this. (Updated the original post with a screenshot.)

**Reply by brandonkelly** [reply]
Craft 4.4 is out with those changes!

**Reply by jacobgraf** [reply]
Just installed and played with them! Thanks again!! 👏🏻

**Comment by jacobgraf** [comment]
@brandonkelly You the man! Much appreciated! Definitely helps! 👏🏻
